src/utils/TokenUtils.ts

```diff
@@ -0,0 +1,244 @@
+   * @param client The client to use for associating the account with tokens.
+   */
+  public static async associateAccountWithTokens(accountId: AccountId,
+                                                 tokenIds: TokenId[],
+                                                 accountKey: PrivateKey,
+                                                 client: Client): Promise<void> {
+    const signTx = await new TokenAssociateTransaction()
+      .setAccountId(accountId)
+      .setTokenIds(tokenIds)
+      .freezeWith(client)
+      .sign(accountKey);
+
+    const txResponse = await signTx.execute(client);
+    await txResponse.getReceipt(client);
+  }
+
+  /**
+   * Mints the given amount of tokens for the given token.
+   * @param tokenId The token ID to mint.
+   * @param metadata The metadata for the minted tokens.
+   * @param supplyKey The supply key to sign the transaction.
+   * @param client The client to use for minting the tokens.
+   */
+  public static async mintToken(tokenId: TokenId,
+                                metadata: string,
+                                supplyKey: PrivateKey,
+                                client: Client): Promise<TransactionReceipt> {
+    const transaction = new TokenMintTransaction()
+      .setTokenId(tokenId)
+      .setMetadata([Buffer.from(metadata)])
+      .freezeWith(client);
+
+    const signTx = await transaction.sign(supplyKey);
+    const txResponse = await signTx.execute(client);
+    return txResponse.getReceipt(client);
+  }
+
+  /**
+   * Creates a token with the given properties.
+   * @param token The properties of the token to create.
+   * @param client The client to use for creating the token.
+   */
+  public static async createToken(token: ITokenProps, client: Client): Promise<[string, TokenId]> {
+    const transaction = this.getTokenCreateTransaction(token);
+
+    let signTx: TokenCreateTransaction;
+    if (token.adminKey) {
+      const adminKey = PrivateKey.fromStringECDSA(token.adminKey);
+      transaction.setAdminKey(adminKey.publicKey);
+      transaction.freezeWith(client);
+      signTx = await (await transaction.sign(adminKey)).signWithOperator(client);
+    } else {
+      transaction.freezeWith(client);
+      signTx = await transaction.signWithOperator(client);
+    }
+
+    const txResponse = await signTx.execute(client);
+    const receipt = await txResponse.getReceipt(client);
+
+    return [token.tokenSymbol, receipt.tokenId!];
+  }
+
+  /**
+   * Returns the supply key for the given token.
+   *
+   * NOTE: The operator key will be used as a supply key by default,
+   * if the supply key is not provided in the properties
+   *
+   * @param token The properties of the token.
+   * @returns The supply key for the token.
+   */
+  public static getSupplyKey(token: ITokenProps): PrivateKey {
+    // The operator key will be used as supply key if one is not provided
+    if (token.supplyKey) {
+      return PrivateKey.fromStringECDSA(token.supplyKey);
```

Comment:
Here and at other places, what if the user wants to pass ED25519 key ? But we can't be sure what kind of key the user will pass. It's okay this way, just a consideration.